File: app/src/main/java/com/example/bhagavadgita/Adapters/AllChaptersAdapter.java
package com.example.bhagavadgita.Adapters;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.bhagavadgita.Models.Chapters;
import com.example.bhagavadgita.R;
import com.example.bhagavadgita.SingleChapterActivity;

import java.util.List;

public class AllChaptersAdapter extends RecyclerView.Adapter<AllChaptersAdapter.ViewHolder> {

    List<Chapters> list;
    Context cContext;

    public AllChaptersAdapter(List<Chapters> list, Context cContext) {
        this.list = list;
        this.cContext = cContext;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        final View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_chapters, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.tvName.setText(list.get(position).getName());
        holder.tvNameTranslated.setText(list.get(position).getNameTranslated());


        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(cContext, SingleChapterActivity.class);
                intent.putExtra("chapterNumber", list.get(position).getChapterNumber());
                cContext.startActivity(intent);
            }
        });
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        TextView tvName;
        TextView tvNameTranslated;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            tvName = itemView.findViewById(R.id.tvName);
            tvNameTranslated = itemView.findViewById(R.id.tvNameTranslated);
        }
    }
}

File: app/src/main/java/com/example/bhagavadgita/ui/main/adapter/AllChaptersAdapter.kt
package com.example.bhagavadgita.ui.main.adapter

import android.content.Context
import android.content.Intent
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.bhagavadgita.data.model.Chapters
import com.example.bhagavadgita.R
import com.example.bhagavadgita.ui.chapter.view.SingleChapterActivity

class AllChaptersAdapter(var list: ArrayList<Chapters>, var cContext: Context) :
    RecyclerView.Adapter<AllChaptersAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.item_chapters, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.tvName.text = list[position].name
        holder.tvNameTranslated.text = list[position].nameTranslated
        holder.tvChapterNo.text = list[position].chapterNumber.toString()
        holder.itemView.setOnClickListener {
            val intent = Intent(cContext, SingleChapterActivity::class.java)
            intent.putExtra("chapterNumber", list[position].chapterNumber)
            cContext.startActivity(intent)
        }
    }

    override fun getItemCount(): Int {
        return list.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var tvName: TextView
        var tvNameTranslated: TextView
        var tvChapterNo: TextView

        init {
            tvName = itemView.findViewById(R.id.tvName)
            tvNameTranslated = itemView.findViewById(R.id.tvNameTranslated)
            tvChapterNo = itemView.findViewById(R.id.tvChapterNo)
        }
    }

    fun addData(newList: List<Chapters>) {
        list.addAll(newList)
    }
}
